uvicorn/_handlers/http.py

```diff
@@ -56,7 +59,7 @@ async def handle_http(
 
     @task.add_done_callback
     def retrieve_exception(task: asyncio.Task) -> None:
-        exc = task.exception()
+        exc = task.exception() if not task.cancelled() else None
 
         if exc is None:
             return
```

Comment:
Can we do the following instead? It makes it easier to understand. 
```python
exc = task.exception()

if task.cancelled() or exc is None:
```

Reply:
If the task has been cancelled, `task.exception` will raise a CancelledError exception.
https://docs.python.org/3/library/asyncio-task.html#asyncio.Task.exception

Reply:
Yeah, I'm aware. But the conditional will work as intended anyway. 🤔

Reply:
> Can we do the following instead? It makes it easier to understand.
> 
> ```python
> exc = task.exception()
> 
> if task.cancelled() or exc is None:
> ```

Calling `task.exception()` after the task is canceled will directly throw a `CanceledError`, which will cause subsequent code to fail to execute.

Reply:
It will throw exception in `task.exception() `
```bash
INFO:     Started server process [2935281]
INFO:     Waiting for application startup.
INFO:     ASGI 'lifespan' protocol appears unsupported.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
INFO:     127.0.0.1:35362 - "GET /1 HTTP/1.1" 200 OK
Exception in callback <function handle_http.<locals>.retrieve_exception at 0x7f24f7ab7310>
handle: <Handle handle_http.<locals>.retrieve_exception>
Traceback (most recent call last):
  File "uvloop/cbhandles.pyx", line 63, in uvloop.loop.Handle._run
  File "/home/ubuntu/uvicorn/uvicorn/_handlers/http.py", line 63, in retrieve_exception
    exc = task.exception()
asyncio.exceptions.CancelledError
```